docs/nim-cmds/action.md

```diff
@@ -3,20 +3,20 @@
 
 work with actions
 
-* [`nim action create [ACTIONNAME] ACTIONPATH`](#nim-action-create-actionname-actionpath)
+* [`nim action create ACTIONNAME [ACTIONPATH]`](#nim-action-create-actionname-actionpath)
 * [`nim action delete ACTIONNAME`](#nim-action-delete-actionname)
 * [`nim action get ACTIONNAME`](#nim-action-get-actionname)
 * [`nim action invoke ACTIONNAME`](#nim-action-invoke-actionname)
 * [`nim action list [PACKAGENAME]`](#nim-action-list-packagename)
 * [`nim action update ACTIONNAME [ACTIONPATH]`](#nim-action-update-actionname-actionpath)
 
-## `nim action create [ACTIONNAME] ACTIONPATH`
+## `nim action create ACTIONNAME [ACTIONPATH]`
```

Comment:
isn't the path required?

Reply:
there is an undocumented feature which allows you to do `nim action create foo.js` as short hand for `nim action create foo foo.js`.

Reply:
I know about the undocumented feature.   But, my job here is to document undocumented features.  So, I started doing that and got a little hung up on this case and have been vacillating about what to do about it.  

I do not agree that ACTIONPATH is required.   I believe the actual semantics (which take some sentences to explain) is that

- if you give two arguments, the first is the name and the second is the path
- if you give one argument, it might be interpreted as the name (if you supply, e.g., the `--docker` or `--native`, or `--sequence` flags) or it might be interpreted as the path (if you give none of the above and the path thus denoted happens to exist)
- as a consequence, at least one argument is required but we don't know a priori whether that argument is the name or the path until examining other arguments and in fact testing file existence, something that is hard to capture in a syntax diagram

Since the situation is just that hairy, I decided on this pass not to alter the auto-generated diagram (which is too stupid to know about all this stuff and just says that ACTIONNAME is required which is actually what the oclif parsing template says) until I get some inspiration on the right way to present this.